server/game/cards/04-MM/SaurianEgg.js

```diff
@@ -0,0 +1,54 @@
+const Card = require('../../Card.js');
+
+class SaurianEgg extends Card {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            match: this,
+            effect: ability.effects.cardCannot('reap')
+        });
+        this.persistentEffect({
+            match: this,
+            effect: ability.effects.cardCannot('fight')
+        });
+
+        this.omni({
+
+            gameAction: ability.actions.discard(context => ({ target: context.player.deck.slice(0,2) })),
+
+            then: context => ({
+                condition: context.preThenEvents &&
+                    context.preThenEvents.filter(
+                        event => !event.cancelled &&
+                        event.name === 'onCardDiscarded' &&
+                        event.card.type === 'creature' &&
+                        event.card.hasHouse('saurian')
+                    ).length > 0,
+                gameAction: ability.actions.sequentialForEach(context => ({
+
+                    forEach: context.preThenEvents.filter(
+                        event => !event.cancelled &&
+                        event.name === 'onCardDiscarded' &&
+                        event.card.type === 'creature' &&
+                        event.card.hasHouse('saurian')
+                    ).map(event => event.card),
+
+                    action: ability.actions.sequential([
+                        ability.actions.putIntoPlay(),
+                        ability.actions.ready(),
+                        ability.actions.addPowerCounter({ amount: 3 })
+
+                    ])
+                })),
+
+                then: {
+                    gameAction: ability.actions.destroy(context => ({
```

Comment:
this should have `alwaysTriggers: true` no?

Reply:
No, the egg should only be destroyed if there are creatures put into play. 

**The card is here:**
https://decksofkeyforge.com/spoilers/1721066

**The text says:**
The Saurian Egg cannot fight or reap.
Omni: Discard the top 2 cards of your deck. **If you discard any Saurian creatures this way**, put them into play ready, give them three +1 power counters, and destroy Saurian Egg.

If I add "alwaysTrue" then the egg will get destroyed, and test cases fail:
* saurian-egg integration Saurian Egg's ability should discard 2 cards that are not saurian creatures and not be destroyed
* saurian-egg integration Saurian Egg's ability should discard only 1 cards that and not be destroyed, when only 1 non-saurian card in the deck.

Let me know if I am miss-understanding your comment, but I think no change is needed, unless:
A) I miss-understand the card.
B) I miss-understood your feedback.

Reply:
I agree that if the egg doesn't discard a saurian creature, the rest of the effect shouldn't trigger (so no `alwaysTriggers: true` in the first `then:`)
However, I'm concerned about the case where the egg discards a saurian creature (e.g. half of Deusillus) and that creature doesn't enter play.  In that case the egg should be destroyed, and I don't think your code will do that

Reply:
Or if it discards a Saurian alpha. Or if you use it and opponent has Ember Imp.

Reply:
Good catch guys. I will update, and add another test.

Reply:
Ember Imp is a red herring - it won't prevent creatures being put into play

Reply:
After chasing my tail for a bit I tried to make a test - but could not find a situation that would make the creatures not playable. So I can't test it, but I have updated the code to support this possibility.